sqlalchemy_jsonapi/declarative/serializer.py

```diff
@@ -0,0 +1,118 @@
+"""A serializer for serializing SQLAlchemy models to JSON API spec."""
+
+import datetime
+
+from inflection import dasherize, underscore
+
+
+class JSONAPISerializer(object):
+    """A JSON API serializer that serializes SQLAlchemy models."""
+
+    def serialize(self, resources):
+        """Serialize resource(s) according to json-api spec."""
+        try:
+            self.fields
+            self.model
+            self.dasherize
+        except AttributeError:
+            raise
+
+        if 'id' not in self.fields:
+            raise ValueError
```

Comment:
I wasn't sure if ValueError or AttributeError should be used here since we are checking is something in in a list. I was thinking ValueError/IndexError but was not sure. Yes we need a message. Something like "'id' must be a field member" or something as such.

We could assume that there will always be a resource.id and leave this out and have an AttributeError raised if 
```
top_level_members['id'] = str(resource.id)
``` 
throws an exception.